### PR TITLE
Feature/create search controller and dto

### PR DESCRIPTION
**Pull request by DmytroTeliukov**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -21,12 +21,29 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-validation</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.28</version>
+        </dependency>
+        <dependency>
```
Comment (malyshevhen) [inline pom.xml:37]: Redundant dependency. Already defined in 'spring-boot-starter-validation'

File: src/main/java/baza/trainee/service/impl/SearchServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import baza.trainee.domain.dto.SearchDto;
+import baza.trainee.service.SearchService;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class SearchServiceImpl implements SearchService {
+    @Override
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/service/impl/SearchServiceImpl.java:12]: It must be an empty string before annotation.

File: src/main/java/baza/trainee/service/SearchService.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package baza.trainee.service;
+
+import baza.trainee.domain.dto.SearchDto;
+
+import java.util.List;
+
+public interface SearchService {
+    /**
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/service/SearchService.java:9]: It must be an empty string before Javadoc comments.

File: pom.xml [hunk 4]
```diff
@@ -21,12 +21,29 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter</artifactId>
         </dependency>
-
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
         <dependency>
```
Comment (malyshevhen) [inline pom.xml:28]: Test dependencies must be at the bottom of the list.

File: src/main/java/baza/trainee/controller/SearchController.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/controller/SearchController.java:34]: Required Javadoc comment

File: src/main/java/baza/trainee/controller/SearchController.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+@RequestMapping("/api/search")
+@RequiredArgsConstructor
+@Validated
+public class SearchController {
+    private final SearchService searchService;
+
+    @GetMapping("/{query}")
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/controller/SearchController.java:51]: Required Javadoc comment

File: src/main/java/baza/trainee/controller/SearchController.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+     *
+     * @param query The search query string to be validated and processed.
+     * @return A list of {@link SearchDto} objects representing
+     * the search results.
+     */
+    @GetMapping("/{query}")
+    List<SearchDto> search(@PathVariable("query")
+                           @NotBlank(message = "Query should not be blank")
+                           @Size(min = 3, max = 120, message = "Size of query should be between 3 and 120 characters")
+                           String query) {
```
Comment (Anatolii-Omelchenko) [inline src/main/java/baza/trainee/controller/SearchController.java]: All input params should be final.

File: src/main/java/baza/trainee/controller/SearchController.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+     * Performs a search operation based on the provided query string.
+     *
+     * @param query The search query string to be validated and processed.
+     * @return A list of {@link SearchDto} objects representing
+     * the search results.
+     */
+    @GetMapping("/{query}")
+    List<SearchDto> search(@PathVariable("query")
+                           @NotBlank(message = "Query should not be blank")
+                           @Size(min = 3, max = 120, message = "Size of query should be between 3 and 120 characters")
```
Comment (Anatolii-Omelchenko) [inline src/main/java/baza/trainee/controller/SearchController.java]: This values (3, 120) should be constants

Comment (Anatolii-Omelchenko) [inline src/main/java/baza/trainee/controller/SearchController.java]: max length of line 80

File: src/main/java/baza/trainee/domain/dto/SearchDto.java [hunk 9]
```diff
@@ -0,0 +1,10 @@
+package baza.trainee.domain.dto;
+
+import baza.trainee.domain.enums.ContentType;
+
+public record SearchDto(
+    String id,
+    String title,
+    String description,
+    ContentType contentType
+) {}
```
Comment (Anatolii-Omelchenko) [inline src/main/java/baza/trainee/domain/dto/SearchDto.java]: The file must end with an empty line.

Comment (Anatolii-Omelchenko) [inline src/main/java/baza/trainee/domain/dto/SearchDto.java]: After '{' should be space.

File: src/main/java/baza/trainee/domain/enums/ContentType.java [hunk 10]
```diff
@@ -0,0 +1,5 @@
+package baza.trainee.domain.enums;
+
+public enum ContentType {
```
Comment (Anatolii-Omelchenko) [inline src/main/java/baza/trainee/domain/enums/ContentType.java:8]: Missed Javadoc.

**Review by malyshevhen — APPROVED** [review]

**Pull request merged**